optuna/samplers/_gp/sampler.py

```diff
@@ -214,42 +228,83 @@ def sample_relative(
+                gp.fit_kernel_params(
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, i],
+                    is_categorical=is_categorical,
+                    log_prior=self._log_prior,
+                    minimum_noise=self._minimum_noise,
+                    initial_kernel_params=cache,
+                    deterministic_objective=self._deterministic,
+                )
+            )
+        self._kernel_params_cache_list = kernel_params_list
 
         best_params: np.ndarray | None
         if self._constraints_func is None:
-            acqf_params = acqf.create_acqf_params(
-                acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
-                kernel_params=kernel_params,
-                search_space=internal_search_space,
-                X=normalized_params,
-                Y=standardized_score_vals,
-            )
-            best_params = normalized_params[np.argmax(standardized_score_vals), :]
+            if n_objectives == 1:
+                assert len(kernel_params_list) == 1
+                acqf_params = acqf.create_acqf_params(
+                    acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
+                    kernel_params=kernel_params_list[0],
+                    search_space=internal_search_space,
+                    X=normalized_params,
+                    Y=standardized_score_vals[:, 0],
+                )
+                best_params = normalized_params[np.argmax(standardized_score_vals), np.newaxis]
+            else:
+                acqf_params_for_objectives = []
+                for i in range(n_objectives):
+                    acqf_params_for_objectives.append(
+                        acqf.create_acqf_params(
+                            acqf_type=acqf.AcquisitionFunctionType.LOG_EHVI,
+                            kernel_params=kernel_params_list[i],
+                            search_space=internal_search_space,
+                            X=normalized_params,
+                            Y=standardized_score_vals[:, i],
+                        )
+                    )
+                acqf_params = acqf.MultiObjectiveAcquisitionFunctionParams.from_acqf_params(
+                    acqf_params_for_objectives=acqf_params_for_objectives,
+                    Y=standardized_score_vals,
+                    n_qmc_samples=128,  # NOTE(nabenabe): The BoTorch default value.
+                    qmc_seed=self._rng.rng.randint(1 << 30),
+                )
+                pareto_params = normalized_params[
+                    _is_pareto_front(-standardized_score_vals, assume_unique_lexsorted=False)
+                ]
+                n_pareto_sols = len(pareto_params)
+                # TODO(nabenabe): Verify the validity of this choice.
+                size = min(self._n_local_search // 2, n_pareto_sols)
+                chosen_indices = self._rng.rng.choice(
+                    np.arange(n_pareto_sols), size=size, replace=False
```

Comment:
This would be sufficient:
```suggestion
                    n_pareto_sols, size=size, replace=False
```